src/core/store/modules/vaults/vaults.actions.ts

```diff
@@ -174,16 +176,35 @@ const getExpectedTransactionOutcome = createAsyncThunk<
 /*                             Transaction Methods                            */
 /* -------------------------------------------------------------------------- */
 
-const approveDeposit = createAsyncThunk<void, { vaultAddress: string; tokenAddress: string }, ThunkAPI>(
+const approveDeposit = createAsyncThunk<
+  { amount: string; spenderAddress: string },
+  { vaultAddress: string; tokenAddress: string },
+  ThunkAPI
+>(
   'vaults/approveDeposit',
-  async ({ vaultAddress, tokenAddress }, { dispatch, getState }) => {
-    const { vaults } = getState();
+  async ({ vaultAddress, tokenAddress }, { dispatch, getState, extra }) => {
+    const { vaults, wallet, network } = getState();
     const vaultData = vaults.vaultsMap[vaultAddress];
-    const isZapin = vaultData.tokenId !== tokenAddress;
-    const spenderAddress = isZapin ? getConfig().CONTRACT_ADDRESSES.zapIn : vaultAddress;
+    const { vaultService, transactionService } = extra.services;
+    const amount = extra.config.MAX_UINT256;
+
+    const accountAddress = wallet.selectedAddress;
+    const isZapin = vaultData.token === tokenAddress;
```

Comment:
should this be `!==`?

Reply:
fixed with 020ad2bba3da6fa711b571e9e3cbee69c0e3b331